src/ui/components/ActionPanel/components/EntryPanel/EntryPanel.tsx

```diff
@@ -11,8 +11,8 @@ import {Dispatch, bindActionCreators} from 'redux';
 import {ActionPanelQA} from 'shared';
 import {DatalensGlobalState, EntryDialogues, sdk} from 'ui';
 import {registry} from 'ui/registry';
-import {addWorkbookInfo, resetWorkbookPermissions} from 'units/workbooks/store/actions';
```

Comment:
It would be great to extend the `addWorkbookInfo` functionality instead of replacing it with `getWorkbook` and `getWorkbookBreadcrumbs`.

Replace `workbooksNames` in `workbooks` reducer with:
```typescript
workbookBreadcrumbsHash: {
	[workbookId]: {
		workbook: WorkbookWithPermissions;
		breadcrumbs: GetCollectionBreadcrumbsResponse;
	}
}
```

And put each new workbook to this hash on demand.